canonical/ubuntu-advantage-client
# proxy: validate urls before setting as proxies (SC-170)

## Proposed Commit Message


> proxy: validate urls before setting as proxies
>
> Introduces a proxy validation function that checks two things:
>
> 1. if the proxy value is a parseable url with http or https scheme
> 2. if the proxy can be used to reach important hardcoded destinations
> 
> If either check fails, the current operation is aborted and an error
message is printed.



## Test Steps

1. Configure ua to use an invalid url as a proxy. Ensure enabling any service fails.
2. Configure ua to use a valid url that doesn't point to a functioning proxy. Ensure enabling any service fails.
3. Configure ua to use a valid url that points to a functioning proxy. Ensure enabling any service succeeds.

## Desired commit type::


## Checklist:
